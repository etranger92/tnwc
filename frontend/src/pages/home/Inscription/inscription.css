@import "../../../style/share.css";
@import "../../../style/animations.css";

:root {
  --padding-label-input-inscription: 0.3rem;
  --padding-inscription-inscritpion: 2rem;
  --padding-paragraph: 0.5rem;
  --padding-icon-media: 1rem;
  --height-label: 10vh;
  --height-paragraph-error: 3vh;
  --font-size-paragraph: 13px;
  --position-top-span: 30%;
  --position-left-span: 5px;
  --span-effect-top: 5%;
  --span-effect-left: 1%;
  --span-effect-fontSize: 13px;
  --width-submit: 85%;
  --height-submit: 4vh;
  --margin-top-facebook-google: 0.5rem;
}

#inscription h1 {
  text-align: center;
}

#inscription h1 span {
  border-bottom: 1px solid var(--txt-color);
}

#flip-card {
  position: relative;
  width: 100%;
  margin: 0 auto;
  padding: .1rem;
}
.css-1wa3eu0-placeholder {
  opacity: 0;
}


#cover-form {
  position: absolute;
  top: 0;
  height: 100%;
  width: 100%;
  background-color: whitesmoke;
  left: 0;
  transition: all .5s;
  display: flex;
  flex-direction: column;
}

#form-inscription {

  display: flex;
  flex-direction: column;
  position: relative;
  z-index: -3;
  transform: rotateY(180deg);
  background-color: white;
  cursor: pointer;
  transition: all .5s;
}

#cover-form span {
  color: var(--txt-color);
  align-self: flex-end;
  padding: 1.5rem;
  padding-bottom: 1rem;
  font-family: var(--signature);
  font-size: 25px;
}

#flip-card:hover #cover-form {
  transform: rotateY(180deg);
  transform-style: preserve-3d;
  z-index: -3;
  transition: all .5s;
}

#flip-card:hover #form-inscription {
  transform: rotateY(0deg);
  transform-style: preserve-3d;
  z-index: 2;

  transition: all .5s;
}

#flip-card:focus #form-inscription {
  z-index: 2;
}

#cover-form>svg {
  margin: auto;
  height: 15%;
  width: 15%;
}

#cover-form>svg>path {
  animation: fillChange 12s linear infinite;
}

@media only screen and (min-width: 700px) {

  #flip-card {
    width: 80%;
  }

  #form-inscription {
    -webkit-box-shadow: -2px 0px 21px -11px rgba(64, 64, 64, 1);
    -moz-box-shadow: -2px 0px 21px -11px rgba(64, 64, 64, 1);
    box-shadow: -2px 0px 21px -11px rgba(64, 64, 64, 1);
    padding: 2rem;
  }


}

@media only screen and (min-width: 1000px) {

  #form-inscription {
    padding: 3rem;
  }
}

#form-part1,
#form-part2,
#form-part3 {
  display: flex;
  flex-wrap: wrap;
}

#form-part-last {
  display: flex;
  padding-bottom: 0.5rem;
}

#form-part-last .sub_form {
  width: 50%;
  margin: auto;
}

#form-part-last .sub_form input {
  margin-top: 2rem;
  height: 8vh;
  width: 100%;
  border-radius: 10px 10px 10px 10px;
}

.sub_form {
  width: 100%;

}

@media only screen and (min-width: 700px) {
  .sub_form {
    width: 50%;
    padding: 2rem;
  }

}

#form-part3 .sub_form {
  display: flex;
  flex-direction: column;
  margin-top: 1rem;
}

#form-part3 .sub_form label {
  height: auto !important;
  font-family: var(--signature);
  font-size: 30px;
}

#form-part3 .sub_form input {
  height: 30vh;
}

#form-part3 .sub_form select {
  height: 10vh;
  width: 20vw;
  background-color: var(--second-color);
  font-size: 16px;
}

.sub_form label {
  display: flex;
  flex-direction: column;
  height: var(--height-label) !important;
  position: relative;
}

.sub_form p {
  min-height: var(--height-paragraph-error);
  font-size: var(--font-size-paragraph);
  padding-bottom: var(--padding-paragraph);
}

.error_message {
  color: var(--error-color);
}

.sub_form label input {
  height: 100% !important;
  width: 100% !important;
  cursor: text !important;
  padding: var(--padding-label-input-inscription) !important;
  border: 1px solid grey ;
  border-radius: 10px 10px 10px 10px;
  padding-top: .8rem !important;
}

.sub_form label span {
  position: absolute;
  top: var(--position-top-span);
  left: var(--position-left-span);
  pointer-events: none;
  transition: all 0.5s;
  color: var(--third-color);
  z-index:1;
}

.sub_form:nth-last-child() {
  text-align: center;
}

.span_effect {
  top: var(--span-effect-top) !important;
  left: var(--span-effect-left) !important;
  font-size: var(--span-effect-fontSize) !important;
}

#submit-button {
  width: var(--width-submit);
  height: var(--height-submit);
  background-color: var(--second-color);
}

.css-g1d714-ValueContainer {
  height: 9.5vh !important;
 
}
.css-yk16xz-control{
    border:1px solid grey !important;
    border-radius: 10px 10px 10px 10px !important;
}
.css-1hwfws3 {
  height:9vh !important;
}