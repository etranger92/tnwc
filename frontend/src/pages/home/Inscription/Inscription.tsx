import React, { useEffect, useState } from "react";
import InscriptionError from "../../../component/Authentification/result/Error/Error";
import InscriptionSuccess from "../../../component/Authentification/result/Success/Success";
import IconHome from "../../../component/ElementsHTML/Icons/IconHome";
import Loader from "../../../component/ElementsHTML/loader/Loader";
import SelectOption from "../../../component/ElementsHTML/selectOption/SelectOption";
import TextArea from "../../../component/ElementsHTML/textArea/TextArea";
import GooglePlacesAutocomplete from "react-google-places-autocomplete";
import {
  clearErrorNotif,
  invalidInput,
  displayError,
} from "../../../logic/functions/error/invalid";
import {
  serverGet,
  serverPost,
} from "../../../logic/functions/sql/requestServe";
import { TESTS } from "../../../logic/functions/test/test";
import "./inscription.css";
//const development = true;
// eslint-disable-next-line @typescript-eslint/no-var-requires
// eslint-disable-next-line @typescript-eslint/no-var-requires
/*type Tprops = {
  //Submit props for test purpose.
  submit?: any;
};*/

//const URL_ADDRESS = "https://www.thenorthwestcollege.com";
const URL_ADDRESS = "https://www.thenorthwestcollege.co.uk";
//const URL_ADDRESS = "http://localhost:3001";
const Inscription: React.FC /*<Tprops>*/ = (/*props*/) => {
  //const { submit } = props;
  //Will be trigger once the user hit the submit Button.
  const [onKeyPress, setOnKeyPress] = useState<boolean>(false);
  const [key, setKey] = useState("");
  const [classTxt, setClassTxt] = useState<string | null>("");
  const [classSurname, setClassSurname] = useState<string | null>("");
  const [classEmail, setClassEmail] = useState<string | null>("");
  const [classAddress, setClassAddress] = useState<string | null>("");
  const [address, setAddress] = useState({
    label: "",
    value: { description: "ss" },
  });
  const form: HTMLFormElement | any = React.useRef<HTMLFormElement>(null);
  const [submitIndiceResult, setSubmitIndice] = useState<boolean | string>(
    "none"
  );
  useEffect(() => {
    //Do not forget to restrict the api to "none" during production otherwise it won't work.
    const url = URL_ADDRESS + "/google-place";
    //const url = "http://localhost:3001/google-place";
    (async () => {
      const response = await serverGet(url);
      const { data } = response;
      setKey(data);
      return response;
    })();
  }, []);

  /**** EVENTS ****/
  //Display Span effect
  const handleFocus = (event: React.FocusEvent<HTMLFormElement>) => {
    const name: string | null = event.target.getAttribute("name");
    const classEffect = "span_effect";
    switch (name) {
      case "name":
        setClassTxt(classEffect);
        break;
      case "surname":
        setClassSurname(classEffect);
        break;
      case "email":
        setClassEmail(classEffect);
        break;
      case "address":
        setClassAddress(classEffect);
        break;
    }
  };
  //Maintain Span effect if value in
  const handleBlur = (event: React.FocusEvent<HTMLFormElement>) => {
    const value: string | null = event.target.value.trim();
    const id: string | null = event.target.getAttribute("name");
    //If there is a value in the input we keep the class.
    const myClass = value ? "span_effect" : "";
    switch (id) {
      case "name":
        setClassTxt(myClass);
        break;
      case "surname":
        setClassSurname(myClass);
        break;
      case "email":
        setClassEmail(myClass);
        break;
      case "address":
        setClassAddress(myClass);
        break;
    }
  };
  //Assist user with indications
  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (!onKeyPress) return;
    const element = event.currentTarget as HTMLInputElement;
    const value: string | null = element.value;
    const type: string | null = element.getAttribute("type");
    //We check the entry if it respects the regex
    if (value && type) {
      TESTS.isEntryValid(value, type)
        ? clearErrorNotif(event.target)
        : invalidInput(event.target, type);
    }
  };
  /**** CHECKS****/
  const checkEmptyValue = (inputs: any[]) => {
    let indice: boolean;
    let doWeSubmit = true;
    const errorAttribute = "data-state";
    const errorIndice = "invalid";
    // eslint-disable-next-line no-unused-vars
    inputs.map((input: HTMLInputElement) => {
      indice = TESTS.isFieldFilled(input.value);
      if (!indice) {
        displayError(input, errorAttribute, errorIndice);
        displayError(input.previousElementSibling, errorAttribute, errorIndice);
        doWeSubmit = false;
      }
      return input;
    });

    return doWeSubmit;
  };
  const checkFormatRegex = (inputs: any[]) => {
    let indice: boolean;
    let doWeSubmit = true;
    const errorAttribute = "data-state";
    const errorIndice = "invalid";
    //Are any inputs that do not respect the format
    inputs.map((input: HTMLInputElement) => {
      indice = TESTS.isEntryValid(input.value, input.type);
      if (indice) {
        clearErrorNotif(input);
      } else {
        invalidInput(input, input.type);
        displayError(input, errorAttribute, errorIndice);
        doWeSubmit = false;
      }
      return input;
    });

    return doWeSubmit;
  };
  /*Create Object from inputs To Be Sent*/
  const createObject = (inputs: any[]) => {
    const object: any = {};
    inputs.map((input: any) => {
      const key = input.getAttribute("data-name");
      const value = input.value;
      if (value) {
        return (object[key] = value);
      }
    });
    //We add the course selected to the object.
    const selectCourse = document
      .getElementById("select-option-courses")
      ?.textContent?.trim();
    object.interest = selectCourse;
    //We add the address through state "address" from google api
    object.address = address.label;
    // We add message in the text area;
    const selectMessage = document.getElementsByTagName("textarea")[0].value;
    object.message = selectMessage;
    console.log(object);
    return object;
  };
  /* Check inputs form */
  const checkInputsForm = (inputs: any[]) => {
    const areAnyEmptyFields = checkEmptyValue(inputs);
    if (!areAnyEmptyFields) {
      //We will enable handleKeyUp to assist the user with some indications
      setOnKeyPress(true);
      return false;
    }
    const areValuesRespectRegex = checkFormatRegex(inputs);
    if (!areValuesRespectRegex) {
      //We will enable handleKeyUp to assist the user with some indications
      setOnKeyPress(true);
      return false;
    }
    return true;
  };
  /* keep inputs that are compulsory */
  const filterInputs = () => {
    let inputs = [...form.current.getElementsByTagName("input")];
    //We pop "submit" input from inputs
    inputs.pop();
    //We remove address (it's optionnal) and we don't want to use any regex for it. We let google api handle it.
    inputs.pop();
    /*
    or if address is not the last one. look at note *(end of file)
    inputs = inputs.filter((input) => {
      const indice = input.getAttribute("id");
      return indice !== "react-select-2-input";
    });
  */
    return inputs;
  };

  /****SUBMIT ACCOUNT****/
  const submitForm = async (event: { preventDefault: () => void }) => {
    try {
      event.preventDefault();
      //We return only inputs that are not optional for the test.
      const inputs = filterInputs();
      const doWeSubmit: boolean = checkInputsForm(inputs);
      if (!doWeSubmit) return;
      /* Test enzyme mock notification
      if (development) {
        //for test mock (submit props)
        if (submit) {
          submit();
        }
        return;
      }*/
      const datas = createObject(inputs);
      //Don't forget the port in the app server.listen(3001)
      // const url = "http://localhost:3001/inscription";
      const url = URL_ADDRESS + "/inscription";

      //We display the loader while we are reaching our backend.
      setSubmitIndice("pending");
      const postDatas = await serverPost(url, datas);
      const { data } = postDatas;
      const result = data ? true : false;
      setSubmitIndice(result);
      return;
    } catch (err) {
      console.log(err);
      setSubmitIndice(false);
    }
  };
  //If we have the connection to the api is right we return it otherwise we return input library our own input.
  const addAddress = key ? (
    <GooglePlacesAutocomplete
      apiKey={key}
      selectProps={{
        address,
        onChange: setAddress,
      }}
    />
  ) : (
    <input
      type="address"
      name="address"
      onKeyUp={handleKeyUp}
      data-name="address"
      id="react-select-3-input"
    />
  );

  return (
    <section id="inscription">
      <h1>
        <span>Inscription </span>
      </h1>
      {submitIndiceResult === "pending" && <Loader />}
      {submitIndiceResult === false && <InscriptionError />}
      {submitIndiceResult === true && <InscriptionSuccess />}
      {submitIndiceResult === "none" && (
        <div id="flip-card">
          <div id="cover-form">
            <IconHome />
            <span> The North West College.</span>
          </div>
          <form
            onFocusCapture={handleFocus}
            onBlurCapture={handleBlur}
            method="post"
            id="form-inscription"
            ref={form}
          >
            <div id="form-part1">
              <div className="sub_form">
                <label htmlFor="name">
                  <span className={"" + classTxt}>
                    <em>Name</em>{" "}
                  </span>
                  <input
                    type="text"
                    name="name"
                    data-name="name"
                    data-state=""
                    onKeyUp={handleKeyUp}
                  />
                </label>
                <p className="error_message"></p>
              </div>
              <div className="sub_form">
                <label htmlFor="surname">
                  <span className={"" + classSurname}>
                    {" "}
                    <em>Surname</em>
                  </span>
                  <input
                    type="text"
                    name="surname"
                    data-name="surname"
                    data-state=""
                    onKeyUp={handleKeyUp}
                  />
                </label>
                <p className="error_message"></p>
              </div>
            </div>
            <div id="form-part2">
              <div className="sub_form">
                <label htmlFor="email">
                  <span className={"" + classEmail}>
                    {" "}
                    <em>Email</em>
                  </span>
                  <input
                    type="email"
                    name="email"
                    onKeyUp={handleKeyUp}
                    data-state=""
                    data-name="email"
                  />
                </label>
                <p className="error_message"></p>
              </div>
              <div className="sub_form">
                <label
                  htmlFor="address"
                  onFocus={() => setClassAddress("span_effect")}
                  onBlur={() => {
                    const divValue = document.getElementsByClassName(
                      "css-1uccc91-singleValue"
                    )[0] as HTMLDivElement;
                    const classo = divValue ? "span_effect" : "";
                    setClassAddress(classo);
                  }}
                >
                  <span className={"" + classAddress}>
                    {" "}
                    <em>Address (optional)</em>
                  </span>
                  {addAddress}
                </label>
                <p className="error_message"></p>
              </div>
            </div>
            <div id="form-part3">
              <div className="sub_form">
                <label htmlFor="interest"> Show an interest.</label>
                <SelectOption />
              </div>
              <div className="sub_form">
                <label htmlFor="txtArea">Send us a message (optional).</label>
                <TextArea
                  rowsIndice={5}
                  colsIndice={5}
                  placeHolder={"Requirements..."}
                />
              </div>
            </div>
            <div id="form-part-last">
              <div className="sub_form">
                <input type="submit" id="submit-button" onClick={submitForm} />{" "}
              </div>
            </div>
          </form>
        </div>
      )}
    </section>
  );
};

export default Inscription;

/*Note:
At filterInput remplace react-select-2-input with react-select-3-input when using local server. I don't know why but it appears with an id different on localhost and server.
on blur as well
*/
