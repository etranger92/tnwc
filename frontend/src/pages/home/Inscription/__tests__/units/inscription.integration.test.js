import React from "react";
import { mount} from "enzyme";
import Inscription from "../../Inscription.tsx";
import toJson from "enzyme-to-json";
import puppeteer from "puppeteer";
jest.setTimeout(50000);

let browser;
let page;
beforeAll(async (done) => {
browser = await puppeteer.launch({
        headless: true,
        slowMo: 0,
       devtools: true,
    });
        page = await browser.newPage();
         await page.tracing.start({
  path: 'trace.json',
  categories: ['devtools.timeline']
 })
        await page.goto("http://localhost:3000/");
        await page.waitForSelector(".hide_loader");
        done();
});
test("No errors are displayed as data-state is empty", async () => {
  let value;
     try {
     value = await page.$$('[data-state="invalid"]');
     expect(value.length).toBe(0);
    } catch (err) {
        console.log("YOUR ERR", err)
    }
})
describe("We test our error notification", () => {
beforeEach( async () => {
    await page.type('[name="name"]', "$Â£$");
     await page.type('[name="surname"]', "Â£3$8");
     await page.type('[name="email"]', "Jonathan9");
     await page.$eval('#submit-button', el => el.click());
  })

test("As all inputs are wrong we should have 6 data-state invalid and three message error displayed ", async () => {
  let value;
     try {
    // value = await page.$$eval('input', el => el.dataset-state == "invalid");
       value = await page.$$('[data-state="invalid"]');
       expect(value.length).toBe(6);
       let text = await page.$$eval('.error_message', elems => elems.map(elem =>  elem.textContent));
    //We have only three inputs that are compulsory for checks. So, address will be returned as null we need to get rid of it.
       text = text.filter(elem => elem.length);
       expect(text.length).toBe(3);
    } catch (err) {
        console.log("YOUR ERR", err)
    }
})
})
afterAll(async (done) => {
 await browser.close();
  done();
})

let wrapper;
describe("We test our inputs of inscription", () => {
beforeEach(() => {
  wrapper = mount(<Inscription />)
   // mockSubmit = jest.fn();
   // wrapper = mount(<Inscription submit={mockSubmit} />);
  });
  test("Snapshot inscription", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("we test that our inputs are correctly filled", () => {
    wrapper.find('[name="name"]').instance().value = "random";
    expect(wrapper.find('[name="name"]').instance().value).toEqual("random");
    wrapper.find('[name="surname"]').instance().value = "random surname";
    expect(wrapper.find('[name="surname"]').instance().value).toEqual("random surname");
    wrapper.find('[name="email"]').instance().value = "random@gmail.com";
    expect(wrapper.find('[name="email"]').instance().value).toEqual(
      "random@gmail.com"
    );
  });
  })


/* The following does not work using enzyme...*/
/*

beforeEach(() => {
  wrapper = mount(<Inscription />)
   // mockSubmit = jest.fn();
   // wrapper = mount(<Inscription submit={mockSubmit} />);
  });
describe("We test our submit form", () => {
 it("all inputs are filled and respect regex; submit is called", () => {
    wrapper.find('[name="name"]').instance().value = "random";
wrapper.find('[name="surname"]').instance().value = "random";
    wrapper.find('[name="email"]').instance().value = "random@gmail.com";
    wrapper.find("#submit-button").simulate("click");
    expect(mockSubmit).toHaveBeenCalled();
  });
 it("all inputs are not filled; submit is not called", () => {
    wrapper.find('[name="name"]').instance().value = "";
    wrapper.find('[name="surname"]').instance().value = "random";
    wrapper.find('[name="email"]').instance().value = "random@gmail.com";
    wrapper.find("#submit-button").simulate("click");
    expect(mockSubmit).not.toHaveBeenCalled();
  });
 it("all inputs are filled but regex are not respected; submit is not called", () => {
    wrapper.find('[name="name"]').instance().value = "sdsd";
    wrapper.find('[name="surname"]').instance().value = "random";
    wrapper.find('[name="email"]').instance().value = "random";
    wrapper.find("#submit-button").simulate("click");
    expect(mockSubmit).not.toHaveBeenCalled();
  });

})
*/

/*//let mockSubmit;
let wrapper;

describe("We test our inputs of inscription", () => {
beforeEach(() => {
  wrapper = mount(<Inscription />)
   // mockSubmit = jest.fn();
   // wrapper = mount(<Inscription submit={mockSubmit} />);
  });
  test("Snapshot inscription", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("we test that our inputs are correctly filled", () => {
    wrapper.find('[name="name"]').instance().value = "random";
    expect(wrapper.find('[name="name"]').instance().value).toEqual("random");
    wrapper.find('[name="surname"]').instance().value = "random surname";
    expect(wrapper.find('[name="surname"]').instance().value).toEqual("random surname");
    wrapper.find('[name="email"]').instance().value = "random@gmail.com";
    expect(wrapper.find('[name="email"]').instance().value).toEqual(
      "random@gmail.com"
    );
  });
  afterEach(() => {
    wrapper = unmount(<Inscription />);
  });
  })*/