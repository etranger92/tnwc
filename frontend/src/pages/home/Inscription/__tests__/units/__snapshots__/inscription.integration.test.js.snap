// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We test our inputs of inscription Snapshot inscription 1`] = `
<Inscription>
  <section
    id="inscription"
  >
    <h1>
      <span>
        Inscription 
      </span>
    </h1>
    <div
      id="flip-card"
    >
      <div
        id="cover-form"
      >
        <IconHome>
          <svg
            className="icon_home"
            height="1062pt"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 1018 1062"
            width="1018pt"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              className=" icon_home_color"
              d="m8020 10341c-41-20-127-65-190-98s-198-103-300-156c-102-52-232-119-290-149-58-31-179-94-270-140-91-47-226-117-300-155-218-113-423-219-610-316-96-49-227-117-290-150-63-32-194-100-290-150-96-49-227-117-290-150-63-32-198-102-300-154-102-53-228-118-280-146s-116-61-141-74c-55-28-90-65-112-118-48-116-130-314-142-340-12-29-79-189-192-460-30-71-74-177-98-235s-69-166-100-240c-92-222-140-336-214-510-38-91-89-214-114-275-79-193-144-348-172-409-26-58-27-59-44-40-18 18-345 418-443 541-67 84-80 97-111 111-15 6-27 16-27 21s-7 17-16 25c-14 14-118 16-997 16-955 0-983-1-994-19-7-11-40-115-74-233-69-238-65-225-147-508-33-113-94-324-136-470-43-146-85-292-96-325-10-33-32-111-50-173-30-108-30-113-14-141 9-15 30-36 46-45 41-25 69-12 185 82 86 69 726 578 761 606 21 15 6 48 244-534 55-135 140-344 190-465 49-121 128-314 175-430 46-115 91-223 98-240 8-16 48-115 90-220 42-104 85-208 94-230 10-22 26-60 36-85 41-97 51-103 226-150 273-72 421-112 524-140 61-17 151-41 200-54 50-13 167-44 260-70 201-54 749-202 1280-346 209-56 542-146 740-200 198-53 529-143 735-198 206-56 473-128 593-161 121-32 223-63 228-68 9-9 32-86 208-713 51-179 96-337 102-352 23-56-80-24 1381-437 444-126 468-132 495-118 36 19 57 57 51 94-4 28-321 504-2063 3103-787 1175-906 1351-988 1473-48 70-87 136-87 146s46 126 101 257c126 297 263 621 367 869 93 221 236 558 367 870 51 121 137 324 190 450 54 127 139 329 190 450s130 308 175 415 131 310 190 450c60 140 145 343 190 450s140 330 210 495c274 643 266 617 209 674-17 17-40 31-52 31s-56-18-97-39z"
              transform="matrix(.1 0 0 -.1 0 1062)"
            />
          </svg>
        </IconHome>
        <span>
           The North West College.
        </span>
      </div>
      <form
        id="form-inscription"
        method="post"
        onBlurCapture={[Function]}
        onFocusCapture={[Function]}
      >
        <div
          id="form-part1"
        >
          <div
            className="sub_form"
          >
            <label
              htmlFor="name"
            >
              <span
                className=""
              >
                <em>
                  Name
                </em>
                 
              </span>
              <input
                data-name="name"
                data-state=""
                name="name"
                onKeyUp={[Function]}
                type="text"
              />
            </label>
            <p
              className="error_message"
            />
          </div>
          <div
            className="sub_form"
          >
            <label
              htmlFor="surname"
            >
              <span
                className=""
              >
                 
                <em>
                  Surname
                </em>
              </span>
              <input
                data-name="surname"
                data-state=""
                name="surname"
                onKeyUp={[Function]}
                type="text"
              />
            </label>
            <p
              className="error_message"
            />
          </div>
        </div>
        <div
          id="form-part2"
        >
          <div
            className="sub_form"
          >
            <label
              htmlFor="email"
            >
              <span
                className=""
              >
                 
                <em>
                  Email
                </em>
              </span>
              <input
                data-name="email"
                data-state=""
                name="email"
                onKeyUp={[Function]}
                type="email"
              />
            </label>
            <p
              className="error_message"
            />
          </div>
          <div
            className="sub_form"
          >
            <label
              htmlFor="address"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <span
                className=""
              >
                 
                <em>
                  Address (optional)
                </em>
              </span>
              <input
                data-name="address"
                id="react-select-3-input"
                name="address"
                onKeyUp={[Function]}
                type="address"
              />
            </label>
            <p
              className="error_message"
            />
          </div>
        </div>
        <div
          id="form-part3"
        >
          <div
            className="sub_form"
          >
            <label
              htmlFor="interest"
            >
               Show an interest.
            </label>
            <SelectOption>
              <div
                id="select-option"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  id="select-option-courses"
                >
                  <span>
                     
                    QTLS
                  </span>
                  <span>
                     
                    <FaAngleDown>
                      <IconBase
                        attr={
                          Object {
                            "viewBox": "0 0 320 512",
                          }
                        }
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                          viewBox="0 0 320 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                            key="0"
                          />
                        </svg>
                      </IconBase>
                    </FaAngleDown>
                  </span>
                </div>
                <ul
                  data-name="selection"
                  id="option-course-container"
                >
                  <li
                    className="
          "
                    key="option0"
                    onClick={[Function]}
                  >
                     
                    <span>
                      AET
                    </span>
                  </li>
                  <li
                    className="
          "
                    key="option1"
                    onClick={[Function]}
                  >
                     
                    <span>
                      CET
                    </span>
                  </li>
                  <li
                    className="
          "
                    key="option2"
                    onClick={[Function]}
                  >
                     
                    <span>
                      DET
                    </span>
                  </li>
                  <li
                    className="
          "
                    key="option3"
                    onClick={[Function]}
                  >
                     
                    <span>
                      CHILD CARE (LVL 2)
                    </span>
                  </li>
                  <li
                    className="
          "
                    key="option4"
                    onClick={[Function]}
                  >
                     
                    <span>
                      CHILD CARE (LVL 3)
                    </span>
                  </li>
                </ul>
              </div>
            </SelectOption>
          </div>
          <div
            className="sub_form"
          >
            <label
              htmlFor="txtArea"
            >
              Send us a message (optional).
            </label>
            <TextArea
              colsIndice={5}
              placeHolder="Requirements..."
              rowsIndice={5}
            >
              <textarea
                cols={5}
                data-name="txtarea"
                placeholder="Requirements..."
                rows={5}
              />
            </TextArea>
          </div>
        </div>
        <div
          id="form-part-last"
        >
          <div
            className="sub_form"
          >
            <input
              id="submit-button"
              onClick={[Function]}
              type="submit"
            />
             
          </div>
        </div>
      </form>
    </div>
  </section>
</Inscription>
`;
